tests/test_time_constrained.py

```diff
@@ -180,3 +182,59 @@ def test_tcpwer_input_formats():
     assert r1.error_rate == r3.error_rate
     assert r1.error_rate == r4.error_rate
     assert r1.error_rate == r5.error_rate
+
+
+def test_time_constrained_sorting_options():
+    from meeteval.wer.wer.time_constrained import time_constrained_minimum_permutation_word_error_rate, \
+        TimeMarkedTranscript
+
+    r1 = TimeMarkedTranscript(['a b', 'c d'], [(0, 1), (0, 1)])
+
+    # "True" checks whether word order matches the word-level timestamps.
+    # Here, it doesn't match, so ValueError is raised.
+    with pytest.raises(ValueError):
+        time_constrained_minimum_permutation_word_error_rate(
+            [r1], [r1], reference_sort=True, hypothesis_sort=True
+        )
+
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r1], reference_sort='word', hypothesis_sort='word'
+    )
+    assert er.error_rate == 0
+
+    r1 = TimeMarkedTranscript(['a b c d', 'e f g h'], [(0, 4), (2, 6)])
+    r2 = TimeMarkedTranscript(['a b c d e f g h'], [(0, 6)])
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r2], reference_sort='word',
+    )
+    assert er.error_rate == 0.75
+
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r2], reference_sort='segment',
+    )
+    assert er.error_rate == 0.75
+
+    # With collar: "segment" keeps word order, so the error becomes 0
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r2], reference_sort='segment', collar=1
+    )
+    assert er.error_rate == 0
+
+    # With collar: "word" does not keep word order, so the overlap gets penalized
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r2], reference_sort='word', collar=1
+    )
+    assert er.error_rate == 0.25
+
+    # False means the user provides the sorting, so we can pass anything
+    r1 = TimeMarkedTranscript(['e f g h', 'a b c d'], [(4, 7), (0, 3)])
+    r2 = TimeMarkedTranscript(['a b c d e f g h'], [(0, 7)])
+    er = time_constrained_minimum_permutation_word_error_rate(
+        [r1], [r2], reference_sort='segment',
+    )
+    assert er.error_rate == 0.5
```

Comment:
I don't get, why the error rate is 0.5.

Reply:
Agreed, it's hard to see because the timestamps result in numbers with many decimal points, and no collar is set. I changed the time stamps so that the error rate becomes 0

Reply:
Yes, it would be good to have extreme cases, such that it is obvious, when two words don't overlap.